website/content/functional-database.md

```diff
@@ -0,0 +1,299 @@
+they are evaluated. So let's give the compiler the authority to make an informed
+guess about _when_ and make `my_var` and `my_function()` semantically
+equivalent.
+
+[1]: Let's say that conventional languages are first 15 from this list:
+https://survey.stackoverflow.co/2022/#most-popular-technologies-language-prof
+
+## Syntax, surly shinier
+
+First of all, lets change the function call to this:
+
+```
+(my_function arg1 arg2 arg3)
+```
+
+It may look strange because function name is within the parenthesis and there is
+no commas between arguments. But trust me, there are benefits to this syntax.
+
+Firstly, when the call has no arguments, the parenthesis can be omitted:
+
+```
+(my_function) == my_function
+```
+
+which feels very natural because of the similar behavior with expressions. As
+intended, function call with no argument and a plain reference to the function
+are expressed with the same syntax.
+
+Let's extend this behavior and allow allow bare function calls in a few places
+where they cannot become ambiguous:
+
+```
+# a list:
+[my_function arg1 arg2 arg3, my_function arg4 arg5 arg6]
+# a declaration:
+let res = my_function arg1 arg2 arg3
+```
+
+Secondly, I'd argue that currying looks natural.
+
+```
+let curry = my_function arg1 arg2
+let res = curry arg3
+
+# or inline:
+let res = (my_function arg1 arg2) arg3
+```
+
+Now let's go one step further and introduce a "pipe" operator. It applies left
+operand as an argument to the right operand:
+
+```
+arg3 | my_function arg1 arg2
+```
+
+This is especially useful for chaining function calls. If we declare `div`,
+`floor` and `mul` as common arithmetic functions, the function call syntax
+really starts to make sense:
+
+```
+12 | div 5 | floor | mul 5
+```
+
+## Running real relations
+
+Up to this point, we were talking language design without a clear justification.
+Language being cool is fine, but it is much more important that the language is
+actually the right tool for the job.
+
+The job, in the case of PRQL, is querying databases.
+
+Core unit of data here is a relation: an ordered set of rows, each of which
+contains entries for each of the columns.
+
+Queries operate in a static environment that contains references to database
+tables and a library of standard functions.
+
+```
+mod default_db {
+    let albums = ...
+    let artists = ...
+    let tracks = ...
+}
+
+mod std {
+    func from tbl -> ...
+    func select cols tbl -> ...
+    func take range tbl -> ...
+
+    func sum col -> ...
+    func average col -> ...
+}
+```
+
+The exact structure and naming may change, but the important part is that we
+have global immutable variables that can be referenced from pure functions.
+
+PRQL currently supports most of the features described above, focused around the
+unconventional function call syntax.
+
+A basic operation on a relation would be:
+
+```
+(take 3 default_db.albums)
+# of with a pipeline
+(default_db.albums | take 3 )
```

Comment:
As above, I think the `default_db` makes it less clear.

One option would be focus just on `take` above (rather than lots of fucns), explain what it does, and then apply it here?